### PR TITLE
Database for bank

### LINKED ISSUE #37: Initialise database for bank
See [Database Schema](https://github.com/MelvinCou/cash-manager/wiki/Base-de-donn%C3%A9es)

### PR DESCRIPTION
**Pull request by MelvinCou**
- JPA for bank
- Simple unit test of a user
`mvn --projects :database test`
- devcontainer
- `account`, `transaction` and `database` modules
- `compose.yaml` at repo's root for Inteliji + env exemples

~:warning: Accidentally deleted `Server/product/compose.yaml `, do not merge for now.~ It has been corrected.

Closes #37

### CODE REVIEW

**Review by MeloLawson — CHANGES_REQUESTED** [review]
About the organization in the database module, usually in java, repositories and entities are in different packages

File: Server/bank/database/src/main/java/com/cashmanager/server/database/repositories/account/AccountsLogs.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+import lombok.ToString;
+
+import java.time.LocalDateTime;
+import java.util.UUID;
+
+@Getter
+@Setter
+@ToString
+@Entity(name = "accounts_logs")
+public class AccountsLogs {
```
[the remark this answers is not in the document]
Comment (MelvinCou) [reply Server/bank/database/src/main/java/com/cashmanager/server/database/repositories/account/AccountsLogs.java]: It seems to be hard to implement :
- logback starts before spring, but requires database available when started. So postgres must run separately.
- logback also logs spring logs, which is very verbose.
- Add custom fields in database is complicated.

File: Server/bank/database/src/main/java/com/cashmanager/server/database/entities/Account.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+@Getter
+@Setter
+@ToString
+@AllArgsConstructor
+@RequiredArgsConstructor
+@Entity(name = "accounts")
+public class Account {
+    @Id
+    private UUID id;
```
Comment (MeloLawson) [inline Server/bank/database/src/main/java/com/cashmanager/server/database/entities/Account.java:21]: You can choose a auto-generate strategy for the primary key in the database
```suggestion
@GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;
```

File: Server/bank/database/src/main/java/com/cashmanager/server/database/entities/AccountsLogs.java [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+import java.time.LocalDateTime;
+import java.util.UUID;
+
+@Getter
+@Setter
+@ToString
+@Entity(name = "accounts_logs")
+public class AccountsLogs {
+    @Id
+    private UUID id;
```
Comment (MeloLawson) [inline Server/bank/database/src/main/java/com/cashmanager/server/database/entities/AccountsLogs.java:17]: same remark for the auto-generation of the primary key

**Review by MeloLawson — CHANGES_REQUESTED** [review]

**Review by MansourWolou — APPROVED** [review]
didn't read all but ok :+1:

**Review by MeloLawson — APPROVED** [review]
ok !

**Pull request merged**